compiler/rustc_passes/src/dead.rs

```diff
@@ -792,7 +812,10 @@ fn check_trait_item(
     id: hir::TraitItemId,
 ) {
     use hir::TraitItemKind::{Const, Fn};
-    if matches!(tcx.def_kind(id.owner_id), DefKind::AssocConst | DefKind::AssocFn) {
+    if matches!(
+        tcx.def_kind(id.owner_id),
+        DefKind::AssocConst | DefKind::AssocTy | DefKind::AssocFn
```

Comment:
This condition is redundant, no?
Trait items cannot have any other `DefKind` than these.